src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -83,10 +88,18 @@ private IFolderItem GetFileSource(string key)
         /// Add path to the configuration
         /// </summary>
         /// <param name="key">App settings key name that have the connections string</param>
-        /// <param name="connectionString">Connection string to access SAAS via ApiHub. <seealso cref="ApiHubJobHostConfigurationExtensions.GetApiHubProviderConnectionStringAsync"/></param>
-        public void AddKeyPath(string key, string connectionString)
+        /// <param name="connectionString">Connection string to access SAAS via ApiHub. <seealso cref="ApiHubJobHostConfigurationExtensions.GetApiHubProviderConnectionStringAsync" /></param>
+        /// <param name="writer">The writer.</param>
+        public void AddKeyPath(string key, string connectionString, TraceWriter writer = null)
         {
-            var root = ItemFactory.Parse(connectionString);
+            ApiHubLogger logger = null;
+
+            if (writer != null)
+            {
+                logger = new ApiHubLogger(writer);
```

Comment:
This is the only place I have access to TraceWriter in this class. this is how this is being called once in ApiHubBindingMetadata in the script repo: (Part of the next PR)

        public override void ApplyToConfig(JobHostConfigurationBuilder configBuilder)
        {
            if (configBuilder == null)
            {
                throw new ArgumentNullException("configBuilder");
            }

            var apiHubConfig = configBuilder.ApiHubConfiguration;

            string connectionString = null;
            if (!string.IsNullOrEmpty(this.Connection))
            {
                connectionString = Utility.GetAppSettingOrEnvironmentValue(Connection);
            }

            FileTraceWriter traceWriter = null;
            foreach (var tracer in configBuilder.Config.Tracing.Tracers)
            {
                traceWriter = tracer as FileTraceWriter;

                if (traceWriter != null)
                {
                    break;
                }
            }

            apiHubConfig.AddKeyPath(this.Key, connectionString, traceWriter);
        }